### DISCUSSION TITLE
How to use keybinds with non-english layout?

### DISCUSSION THREAD

**Discussion by khavronin-artem** [jarun/nnn#1892] [Q&A]
It simply doesnt work. Changing keybindings in nnn.h doesn't work because of C language.

**Comment by 7ocb** [comment]
There is also problem I faced when I initially tried to modify bindings to support non-english layout is that non-english chars are multi-byte and nnn does not support (at least did not at that time) multi-byte chars in bindings.

Curious thing that I just checked and Midnight Commander somehow supports these bindings (i.e. "next" binding which is Control+N works both with english and russian layours), so may have sense to look how they implemented this.

**Reply by 7ocb** [reply]
I just look via `xev` tool and it seems that "Ctrl+KEY" sends english keycode for this KEY even if non-english layout is active.

**Comment by jarun** [comment]
Sorry, I don't have a non-English keyboard.
Can you please raise a PR?

**Reply by jarun** [reply]
Adding @N-R-K @luukvbaal @KlzXS

**Reply by N-R-K** [reply]
I also don't use non-english keyboard and have no clue what that entails :man_shrugging:

**Reply by jarun** [reply]
Looks like MC uses glib.
